nemo_curator/filters/doc_filter.py

```diff
@@ -81,6 +81,15 @@ def keep_document(self, scores: Any) -> bool:
             "keep_document method must be implemented by subclasses"
         )
 
+    def get_backend(self) -> str:
```

Comment:
```suggestion
    def get_backend(self) -> Literal["pandas", "cudf", "any"]:
```

(what's any though?)

Reply:
Seems like this could be an [abstractproperty](https://docs.python.org/3/library/abc.html#abc.abstractproperty) (now abstractmethod + property). 

And instead of the abstract class defining it, IMO the inherited class should explicitly define that property to be `pandas` or whatever

Reply:
"any" means that the module can operate on either "pands" or "cudf" as a backend. Will update to be a property instead of a method.

> And instead of the abstract class defining it, IMO the inherited class should explicitly define that property to be pandas or whatever

The reason that I have the base class start off with a default of `pandas` is that right now I believe every DocumentFilter has a backend of pandas. In my mind, it's more convenient to set the default and have a few classes override it in the future if they want. What do you think?